src/main/scala/com/databricks/spark/xml/util/TypeCast.scala

```diff
@@ -115,11 +115,14 @@ private[xml] object TypeCast {
   )
 
   private def parseXmlTimestamp(value: String, options: XmlOptions): Timestamp = {
+    val timeZone = options.timeZone
     val formatters = options.timestampFormat.map(DateTimeFormatter.ofPattern).
       map(supportedXmlTimestampFormatters :+ _).getOrElse(supportedXmlTimestampFormatters)
     formatters.foreach { format =>
       try {
-        return Timestamp.from(ZonedDateTime.parse(value, format).toInstant)
+        return Timestamp.from(
+          ZonedDateTime.parse(value, format.withZone(ZoneId.of(timeZone))).toInstant
```

Comment:
The new logic should only be applied to the custom timestamp format, not the built-in ones.
I suppose, if possible, you check to see if the pattern has a TZ and augment it with the TZ specified in `spark.sql.session.timeZone` if not.